### ISSUE TITLE
Consider adding support for TypedSQL

### ISSUE THREAD

**Issue by infomiho** [wasp-lang/wasp#2268]
Depending on user requests, we could support the TypedSQL Prisma feature released in Prisma 5.19.0

https://www.prisma.io/docs/orm/prisma-client/using-raw-sql/typedsql#getting-started-with-typedsql

**Comment by the-notable** [comment]
Ended up here searching for a way to implement a self-referential recursive query using Prisma in Wasp. Prisma doesn't natively support them, but they can be achieved through the use of Prisma's TypedSQL feature. 

But it seems while Wasp does allow for Prisma Preview Features generally, it doesn't allow for the TypedSQL feature? 

Is it because under the hood Wasp doesn't add the ``--sql`` flag to the ``prisma generate`` command?

**Comment by sodic** [comment]
> Is it because under the hood Wasp doesn't add the --sql flag to the prisma generate command?

Yes, I'm guessing that's the case. Here's the code that calls `prisma generate`: https://github.com/wasp-lang/wasp/blob/499e84b906ad2c2ba07bd8138c8e70b80f670946/waspc/src/Wasp/Generator/DbGenerator/Jobs.hs#L159

I'll tag @infomiho. He's the expert.

Should we maybe support the `--sql` flag by default? Are there any the downsides. @the-notable Feel free to jump in, you likely know more than I do :)
